engine/src/main/java/org/terasology/engine/recording/EventSystemReplayImpl.java

```diff
@@ -303,11 +303,11 @@ private boolean shouldAddToLibrary(Class<? extends Event> eventType) {
     public void registerEventHandler(ComponentSystem handler) {
         Class handlerClass = handler.getClass();
         if (!Modifier.isPublic(handlerClass.getModifiers())) {
-            logger.error("Cannot register handler {}, must be public", handlerClass.getName());
+            logger.atError().addArgument(() -> handlerClass.getName()).log("Cannot register handler {}, must be public");
             return;
         }
 
-        logger.debug("Registering event handler {}", handlerClass.getName());
+        logger.atDebug().addArgument(() -> handlerClass.getName()).log("Registering event handler {}");
```

Comment:
O(1) method call, only done once when registering event handlers
```suggestion
        logger.debug("Registering event handler {}", handlerClass.getName()); //NOPMD
```